Userland/Libraries/LibGfx/Filters/LumaFilter.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022, Tobias Christiansen <tobyase@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "LumaFilter.h"
+namespace Gfx {
+
+void LumaFilter::apply(u8 lower_bound, u8 upper_bound)
+{
+    if (upper_bound < lower_bound)
+        return;
+
+    int height = m_bitmap.height();
+    int width = m_bitmap.width();
+
+    auto format = m_bitmap.format();
+    VERIFY(format == BitmapFormat::BGRA8888 || format == BitmapFormat::BGRx8888);
```

Comment:
Is there a specific reason for verifying this here? It doesn't look like the filter assumes it.

Reply:
It was there before, I only extended it.
The value is used to determine which of the {get,set}_pixel() functions to use.

Those two Formats are afaict the ones passed around by PixelPaint.
We only support both Formats and not the other ones. I don't know enough about what is going on there to also support the others. If anybody needs other formats, they will hit the VERIFY and have to implement it themselves.

Reply:
Sure, that sounds reasonable.